flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecScriptTransform.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.flink.table.planner.delegation.PlannerBase;
+import org.apache.flink.table.planner.plan.nodes.exec.ExecNode;
+import org.apache.flink.table.planner.plan.nodes.exec.ExecNodeBase;
+import org.apache.flink.table.planner.plan.nodes.exec.ExecNodeConfig;
+import org.apache.flink.table.planner.plan.nodes.exec.ExecNodeContext;
+import org.apache.flink.table.planner.plan.nodes.exec.InputProperty;
+import org.apache.flink.table.runtime.script.ScriptTransformIOInfo;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.logical.LogicalType;
+
+import java.lang.reflect.Constructor;
+import java.util.Collections;
+
+/** Batch {@link ExecNode} for ScripTransform. */
+public class BatchExecScriptTransform extends ExecNodeBase<RowData>
+        implements BatchExecNode<RowData> {
+
+    private final int[] inputIndexes;
+    private final String script;
+    private final ScriptTransformIOInfo scriptTransformIOInfo;
+    private final LogicalType inputType;
+    private final String scriptTransformOperatorClassName;
+
+    public BatchExecScriptTransform(
+            ReadableConfig tableConfig,
+            InputProperty inputProperty,
+            LogicalType inputType,
+            LogicalType outputType,
+            String description,
+            int[] inputIndexes,
+            String script,
+            ScriptTransformIOInfo scriptTransformIOInfo,
+            String scriptTransformOperatorClassName) {
+        super(
+                ExecNodeContext.newNodeId(),
+                ExecNodeContext.newContext(BatchExecLimit.class),
+                ExecNodeContext.newPersistedConfig(BatchExecLimit.class, tableConfig),
+                Collections.singletonList(inputProperty),
+                outputType,
+                description);
+        this.inputIndexes = inputIndexes;
+        this.script = script;
+        this.inputType = inputType;
+        this.scriptTransformIOInfo = scriptTransformIOInfo;
+        this.scriptTransformOperatorClassName = scriptTransformOperatorClassName;
+    }
+
+    @Override
+    protected Transformation<RowData> translateToPlanInternal(
+            PlannerBase planner, ExecNodeConfig config) {
+        Transformation<RowData> inputTransform =
+                (Transformation<RowData>) getInputEdges().get(0).translateToPlan(planner);
+        OneInputStreamOperator<RowData, RowData> scriptOperator = getScriptTransformOperator();
+        return new OneInputTransformation<>(
+                inputTransform,
+                getDescription(),
+                SimpleOperatorFactory.of(scriptOperator),
+                InternalTypeInfo.of(getOutputType()),
+                inputTransform.getParallelism());
+    }
+
+    private OneInputStreamOperator<RowData, RowData> getScriptTransformOperator() {
+        try {
+            Class<?> cls =
+                    Class.forName(
+                            scriptTransformOperatorClassName,
```

Comment:
We can hard code the `HiveScriptTransformOperator` class name in the static field of this class.